scheduler/src/cook/log_structured.clj

```diff
@@ -42,30 +42,38 @@
   "Logs a structured message at the specified level.
   Supports specifying a custom logger namespace as an optional parameter."
   ([level data metadata]
-   `log/log ~level (json/write-str (assoc ~metadata :data ~data)))
-  ([level data metadata logger-ns]
-   `(log/log ~logger-ns ~level nil (json/write-str (assoc ~metadata :data ~data)))))
+   `(log/log ~level (json/write-str (assoc ~metadata :data ~data))))
```

Comment:
This code here means we may unconditionally process each log message (including super verbose /debug logs. 

My suspicion is that the logging library behind the scenes will check the log level log component and not generate strings that aren't actually logged. We should verify that that assumption holds. E.g., a unit test. 

Structured log something with a function call. Make sure that the function isn't actually invoked --- if it is invoked, then I'd advocate not structuring the /debug logs.)

Reply:
According to the [documentation](https://clojure.github.io/tools.logging/#clojure.tools.logging/log) `log` only evaluates and logs the message is the logger is enabled (`log*` does not check if the logger is enabled).

Reply:
I added a unit test to make sure that the functions aren't called at the debug level.